### ISSUE TITLE
Prune taxa error

### ISSUE THREAD

**Issue by biologyguy** [biologyguy/BuddySuite#110]
```
b12b879e8cd954ef5a4a074b524f8a23
1
# PhyloBuddy: 1.2.5
# Function: prune_taxa
# Python: 3.5.3 |Continuum Analytics, Inc.| (default, Mar  6 2017, 11:58:13) [GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
# Platform: linux
# User: hashless
# Date: 2017-03-08

error: unterminated character set at position 0

  File "PhyloBuddy.py", line 1436, in main
    command_line_ui(*initiation)
  File "PhyloBuddy.py", line 1373, in command_line_ui
    prune_taxa(phylobuddy, *in_args.prune_taxa[0])
  File "PhyloBuddy.py", line 825, in prune_taxa
    if re.search(pattern, taxon):  # Sets aside the names of the taxa to be pruned
  File "re.py", line 173, in search
    return _compile(pattern, flags).search(string)
  File "re.py", line 293, in _compile
    p = sre_compile.compile(pattern, flags)
  File "sre_compile.py", line 536, in compile
    p = sre_parse.parse(p, flags)
  File "sre_parse.py", line 829, in parse
    p = _parse_sub(source, pattern, 0)
  File "sre_parse.py", line 437, in _parse_sub
    itemsappend(_parse(source, state))
  File "sre_parse.py", line 545, in _parse
    source.tell() - here)

//
```

**Comment by KarlKeat** [comment]
A quick Google search tells me this might happen for example if someone uses a pattern with a reserved character in it, like "-". In that case, it would try to take a range, which depending on the characters surrounding it might not be possible. We would need to know the actual query to know for sure. However, this means that we either need to pre-parse regexes that people feed in for special characters, or we need to let the user deal with it and improve exception handling for invalid regular expressions.

**Comment by biologyguy** [comment]
I was thinking it might be useful to add the command used to the metadata header of crash reports. It would help resolve bugs like this.

**Comment by biologyguy** [comment]
There are a ton of [ways for regex to fail](https://bugs.python.org/file38036/re_errors_diff.txt). I'll need to catch a generic exception and handle it gracefully.

**Comment by biologyguy** [comment]
Resolved with commit 1c95dac1be7ba8a5b57ea2901882e42d53316d21
Used a new function in buddy_resources to check all incoming regular expressions before trying to apply them.
Will be released with v1.2.6 very soon
